src/track/beatstats.cpp

```diff
@@ -0,0 +1,55 @@
+#include <QList>
+#include <QMap>
+
+#include <algorithm>
+#include <cmath>
+
+#include "track/beatstats.h"
+
+
+void MovingMode::update(double newValue, double oldValue) {
+    m_tempoFrequency[newValue] += 1;
+    if (not std::isnan(oldValue)) {
+        m_tempoFrequency[oldValue] -= 1;
+    }
+}
+double MovingMode::compute() {
+    QMapIterator<double,  int> tempo(m_tempoFrequency);
+    auto mode = tempo.next();
```

Comment:
No, it doesn't. Not only 2 values but in the more general case, if we have a non-unique mode, it will just return the first ocurrence. I will fix that